src/stream/src/lib.rs

```diff
@@ -55,16 +59,42 @@ pub mod task;
 #[cfg(test)]
 risingwave_expr_impl::enable!();
 
+pub(crate) static CONFIG: OnceLock<Arc<StreamingConfig>> = OnceLock::new();
+
+fn streaming_config() -> &'static StreamingConfig {
+    CONFIG
+        .get()
+        .expect("should be set when compute node initializing")
+}
+
 mod consistency {
     //! This module contains global variables and methods to access the stream consistency settings.
 
     use std::sync::LazyLock;
 
+    use risingwave_common::config::StrictConsistencyOption;
     use risingwave_common::util::env_var::env_var_is_true;
 
     static INSANE_MODE: LazyLock<bool> = LazyLock::new(|| env_var_is_true("RW_ENABLE_INSANE_MODE"));
 
-    pub fn insane() -> bool {
+    pub(crate) fn insane() -> bool {
         *INSANE_MODE
     }
+
+    pub(crate) fn strict_consistency() -> StrictConsistencyOption {
+        crate::streaming_config().strict_consistency
+    }
+
+    macro_rules! inconsistency_panic {
+        ($($arg:tt)*) => {
+            match crate::consistency::strict_consistency() {
+                risingwave_common::config::StrictConsistencyOption::On => {
+                    tracing::error!($($arg)*);
+                    panic!("inconsistency happened, see error log for details");
```

Comment:
I'm not sure if `stdout` or `stderr` can always be flushed in this case. Would you please verify that?

Reply:
Just tested on my local machine, it's flushed